sdk/src/client/secret/ledger_nano.rs

```diff
@@ -398,6 +374,53 @@ impl SignTransactionEssence for LedgerSecretManager {
 
         Ok(Unlocks::new(unlocks)?)
     }
+
+    async fn sign_transaction(
+        &self,
+        prepared_transaction_data: PreparedTransactionData,
+    ) -> Result<Payload, Self::Error> {
+        super::default_sign_transaction(self, prepared_transaction_data).await
+    }
+}
+
+impl SecretManagerConfig for LedgerSecretManager {
+    type Config = bool;
+
+    fn to_config(&self) -> Option<Self::Config> {
+        Some(self.is_simulator)
+    }
+
+    fn from_config(config: &Self::Config) -> Result<Self, Self::Error> {
+        Ok(Self::new(*config))
+    }
+}
+
+/// the Ledger Nano S(+)/X app can present the user a detailed view of the transaction before it
+/// is signed but only with BasicOutputs, without extra-features and if the Essence is not too large.
+/// If criteria are not met, blind signing is needed.
+/// This method finds out if we have to switch to blind signing mode.
+pub fn needs_blind_signing(prepared_transaction: &PreparedTransactionData, buffer_size: usize) -> bool {
+    match &prepared_transaction.essence {
+        TransactionEssence::Regular(essence) => {
+            for output in essence.outputs().iter() {
+                // only basic outputs allowed
+                if let Output::Basic(output) = output {
+                    if output.simple_deposit_address().is_some() {
+                        continue;
+                    }
+                }
+                // not fine, return
+                return true;
+            }
```

Comment:
this loop could be replaced with:
```rust
            if !essence
                .outputs()
                .iter()
                .all(|o| matches!(o, Output::Basic(output) if output.simple_deposit_address().is_some()))
            {
                return true;
            }
```
It's debatable whether that's more readable, but IMHO it is.